### ISSUE TITLE
@capacitor/camera - Android "Selected Photos Access"

### ISSUE THREAD

**Issue by julienkermarec** [ionic-team/capacitor-plugins#2150]
### Plugin
@capacitor/camera

### Description
Android 14 introduces Selected Photos Access, which allows users to grant apps access to specific images and videos in their library, rather than granting access to all media of a given type.

![Capture d’écran 2024-07-11 à 11 05 11](https://github.com/ionic-team/capacitor-plugins/assets/3876561/283f6ae3-8357-4b85-88dd-daa49024d846)

This change is only enabled if your app targets Android 14 (API level 34) or higher. If you don't use the photo picker yet, we recommend [implementing it in your app](https://developer.android.com/training/data-storage/shared/photopicker) to provide a consistent experience for selecting images and videos that also enhances user privacy without having to request any storage permissions.

If you maintain your own gallery picker using storage permissions and need to maintain full control over your implementation, [adapt your implementation](https://developer.android.com/about/versions/14/changes/partial-photo-video-access#app-gallery-picker) to use the new [READ_MEDIA_VISUAL_USER_SELECTED](https://developer.android.com/reference/android/Manifest.permission#READ_MEDIA_VISUAL_USER_SELECTED) permission. If your app doesn't use the new permission, the system runs your app in a [compatibility mode](https://developer.android.com/about/versions/14/changes/partial-photo-video-access#compatibility-mode).

### Platform(s)
Android

### Preferred Solution
Implement the new way to request permission & picture selection

### Alternatives
No other plugins alternative

### Additional Context
New Permission to add 
```
<!-- To handle the reselection within the app on devices running Android 14 or higher if your app targets Android 14 (API level 34) or higher.  -->
<uses-permission android:name="android.permission.READ_MEDIA_VISUAL_USER_SELECTED" />
```

New requesting permission : 
```
// Register ActivityResult handler
val requestPermissions = registerForActivityResult(RequestMultiplePermissions()) { results ->
    // Handle permission requests results
    // See the permission example in the Android platform samples: https://github.com/android/platform-samples
}

// Permission request logic
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
    requestPermissions.launch(arrayOf(READ_MEDIA_IMAGES, READ_MEDIA_VIDEO, READ_MEDIA_VISUAL_USER_SELECTED))
} else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
    requestPermissions.launch(arrayOf(READ_MEDIA_IMAGES, READ_MEDIA_VIDEO))
} else {
    requestPermissions.launch(arrayOf(READ_EXTERNAL_STORAGE))
}

```

**Comment by M13CZ** [comment]
Is there aby workaround for this before it will be fixed on plugin side?

**Comment by julienkermarec** [comment]
> Is there aby workaround for this before it will be fixed on plugin side?

Yes, on can use @capacitor/filesystem for selecting private photo

For using camera, you need to continue using @capacitor/camera

**Comment by jpike88** [comment]
My app just got rejected because of something related to this.

**Comment by eric-horodyski** [comment]
Closing this as part of our ongoing effort to keep Capacitor’s backlog healthy and focused on current development priorities.

More context on this initiative:
[https://ionic.io/blog/keeping-capacitors-backlog-healthy](https://ionic.io/blog/keeping-capacitors-backlog-healthy)

**Additional note:**
We are currently working on a Camera refactor and are taking this into account. Thanks for the issue, and apologies for the delay!
